Automattic/wp-calypso
# State: Modularize Jetpack Connect state

This PR is one of the PRs that aim to modularize the state in Calypso. It is responsible for modularizing Jetpack Connect, a piece that is generally pretty independent from the rest of Calypso.

While it doesn't shave off too much from the main entry point, it's still a good improvement, because only the code that actually uses that state tree will load it, and this moves us a step forward towards a more modular Calypso. 

While the PR looks pretty big, it's very straightforward - it moves all actions and selectors into their dedicated files, moves out the redux tree to be independently initialized only by these actions/selectors, and moves the action type out of the main action types file. I'm happy to split to multiple PRs, but I believe that the atomic commits should make reviewing very straightforward.

For more details on state modularization, see #39261 and p4TIVU-9lM-p2.

#### Changes proposed in this Pull Request

* Add Jetpack Connect reducer initialization
* Initialize Jetpack Connect state in actions
* Initialize Jetpack Connect state in selectors
* Remove Jetpack Connect state from monolithic reducer
* Modularize Jetpack Connect action types
* Modularize Jetpack Connect actions
* Modularize Jetpack Connect selectors

#### Testing instructions

* Checkout this branch.
* Go to http://calypso.localhost:3000/read
* Verify that `state.jetpackConnect` is not defined.
* Go to http://calypso.localhost:3000/jetpack/connect
* Verify that * Verify that `state.jetpackConnect` is defined and you can see the redux state tree.
* Connect a Jetpack site starting from `/jetpack/connect` and verify it works well and produces no errors.
* Connect a Jetpack site starting from wp-admin and verify it works well and produces no errors in Calypso.
